pandas/core/indexes/datetimes.py

```diff
@@ -607,6 +611,9 @@ def _simple_new(cls, values, name=None, freq=None, tz=None,
                            dtype=dtype, **kwargs)
             values = np.array(values, copy=False)
 
+        # values should be a numpy array
```

Comment:
use the format, not commment is needed
``assert ...., "values are not an np.ndarray"``
assert the integer dtype as well

Reply:
What's the intention of an ``assert is_integer_dtype(values)``? ``values`` is often an ndarray of datetime64[ns] at this stage which means this assert fails very often.

Reply:
oh right, sorry the assert should be assert is_datetime64_dtype, it *always* should be this

Reply:
its just better that we have certain guarantees in theses low level constructors